File: implementations/java-springboot/harness/test/testdata/no-cross-bc-domain-import/good/payment/domain/Refund.java
package com.example.accountservice.payment.domain;

public class Refund {
    private String refundId;
    private String paymentId;

    public static Refund request(String paymentId) {
        Refund refund = new Refund();
        refund.paymentId = paymentId;
        return refund;
    }
}

File: implementations/kotlin-springboot/harness/test/testdata/no-cross-aggregate-reference/bad-payment-holds-refund/payment/domain/Refund.kt
package com.example.accountservice.payment.domain

class Refund private constructor() {
    var refundId: String = ""
        private set

    var paymentId: String = ""
        private set

    companion object {
        fun create(paymentId: String): Refund =
            Refund().apply {
                this.refundId = "ref-1"
                this.paymentId = paymentId
            }
    }
}
